src/commands.ts

```diff
@@ -506,12 +508,33 @@ export const insertInlineFootnote =
         footnotesSection.node,
         schema.nodes.footnotes_element
       )
-      // TODO: Revisit this position calculation as it doesn't sound right to always push the note to the end.
+      // Get position for inserting the footnote
+      let index = 0
+      /* 
+      TODO: Would it be faster to use the footnotes plugin state to get the inline-footnotes, 
+      filter them based on the insertedAt position, and then use the length + 1 as the index?
+      */
+      state.doc.slice(0, insertedAt).content.descendants((node) => {
+        if (node.type === schema.nodes.inline_footnote) {
```

Comment:
Don't you need to exclude the table's inline footnotes?

Reply:
Don't you need to exclude rejected inline footnotes?

Reply:
I have to check this...